sources/packages/backend/apps/api/src/route-controllers/restriction/restriction.aest.controller.ts

```diff
@@ -214,6 +219,48 @@ export class RestrictionAESTController extends BaseController {
     }
   }
 
+  /**
+   * Soft deletes a provincial restriction from Student.
+   * @param studentId ID of the student to get a restriction.
+   * @param studentRestrictionId ID of the student restriction to be deleted.
+   * @param payload delete restriction details.
+   */
+  @Roles(Role.StudentDeleteRestriction)
+  @ApiNotFoundResponse({
+    description: "Provincial restriction not found.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description: "Provincial restriction is already set as deleted.",
+  })
+  @Patch("student/:studentId/student-restriction/:studentRestrictionId/delete")
```

Comment:
I can see that existing APIs use student id on APIs that resolve and get restriction detail. 
Are we keeping student id here to follow the existing APIs? or is there a different reason?

Reply:
Following the existing API pattern.